include/game.h

```diff
@@ -261,6 +201,69 @@ typedef enum {
     Game_99 = 99,
 } GameState;
 
+#define STAGE_INVERTEDCASTLE_MASK 0x1F
+#define STAGE_INVERTEDCASTLE_FLAG 0x20
+typedef enum {
```

Comment:
I think the manual numbering here is fine for now although there's various disagreements about style. This PR adds a lot of useful documentation and I wouldn't hold it up for this.